internal/users/users_test.go

```diff
@@ -128,6 +129,53 @@ func TestUpdateLocalGroups(t *testing.T) {
 	}
 }
 
+func TestCleanupSystemGroups(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		groupFilePath string
+
+		getentCmd []string
+
+		wantErr bool
+	}{
+		"Successfully cleans up groups": {groupFilePath: "root_in_groups.group"},
+
+		"Error when NSS errors out":                   {groupFilePath: "root_in_groups.group", getentCmd: []string{"nss-fail"}, wantErr: true},
+		"Error on missing groups file":                {groupFilePath: "does_not_exists.group", wantErr: true},
+		"Error when groups file is malformed":         {groupFilePath: "malformed_file.group", wantErr: true},
+		"Error on any unignored delete gpasswd error": {groupFilePath: "gpasswdfail_in_deleted_group.group", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			destCmdsFile := filepath.Join(t.TempDir(), "gpasswd.output")
+			groupFilePath := filepath.Join("testdata", tc.groupFilePath)
+			cmdArgs := []string{"env", "GO_WANT_HELPER_PROCESS=1",
+				fmt.Sprintf("GO_WANT_HELPER_PROCESS_DEST=%s", destCmdsFile),
+				fmt.Sprintf("GO_WANT_HELPER_PROCESS_GROUPFILE=%s", groupFilePath),
+				os.Args[0], "-test.run=TestMockgpasswd", "--"}
+
+			cleanupOptions := []users.Option{users.WithGpasswdCmd(cmdArgs), users.WithGroupPath(groupFilePath)}
+			if tc.getentCmd != nil {
```

Comment:
You went a little bit the easy path here. I suggest that we really mock getent to be under control what we are testing against. Imagine there is a local user matching one of those user names here on the test machine by accident? The fact that we rely on `root` as being system and always there and the rest not existing doesn’t make your test independent of the machine.

Please take the time to mock getent and control different user scenarios with this.